lib/MusicBrainz/Server/Controller/Admin/Attributes.pm

```diff
@@ -93,12 +95,22 @@ sub edit : Chained('attribute_base') Args(1) RequireAuth(account_admin) {
     my $form = $c->form( form => $form_name, init_object => $attr );
 
     if ($c->form_posted && $form->process( params => $c->req->params )) {
-        $c->model('MB')->with_transaction(sub {
-            $c->model($model)->update($id, { map { $_->name => $_->value } $form->edit_fields });
-        });
-
-        $c->response->redirect($c->uri_for('/admin/attributes', $model));
-        $c->detach;
+        my $old_entity_type = $attr->entity_type;
+        my $new_entity_type = $form->field("entity_type")->value;
+
+        if ($old_entity_type ne $new_entity_type) {
+            $form->field('entity_type')->add_error(
+                l('Entity type \'{entity_type}\' cannot be changed; please delete and readd attribute instead.',
```

Comment:
Hmm, I think it's probably right for them to be localized. Even though all the admins know English, they might have a different language selected in the UI and it would be weird to make an exception here. But I'd avoid adding strings to the admin UI where not necessary, or I'd make them simple if required.